test/models/user_contest_test.rb

```diff
@@ -95,6 +95,24 @@ class UserContestTest < ActiveSupport::TestCase
     assert_equal pucs.find(gold.id).contest_points, 7
   end
 
+  test 'short problem scoring' do
+    c = create(:full_contest, short_problems: 4)
+    ucs = c.user_contests
+    sp1 = c.short_problems.first
+    sp2 = c.short_problems.second
+    sp3 = c.short_problems.third
+    sp4 = c.short_problems.fourth
+    sp3.update(answer: 1)
+    none = ucs.first
+    none.short_submissions.first.update(answer: sp1.answer)
+    none.short_submissions.second.update(answer: nil)
+    none.short_submissions.third.update(answer: '')
+    none.short_submissions.fourth.update(answer: sp4.answer.to_i + 1)
+
+    pucs = ucs.processed
+    assert_equal pucs.find(none.id).short_mark, 1
```

Comment:
Are you not testing for other cases?

Reply:
Can you give me some example?

Reply:
Oh wait I see what you're doing here. So the thing about tests is that you want to try to isolate every feature in separate tests. This is so that we can detect if there is one specific feature that is wrong. In this particular instance it is advisable to have one test for correct answers; one test for wrong answers; and one test for empty answers.